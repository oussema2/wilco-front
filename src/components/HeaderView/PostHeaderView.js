import React from 'react';
import PropTypes from 'prop-types';
import HeaderView from './HeaderView';

const PostHeaderView = ( {
	testID, pilotName, dateToDisplay, optionsOnPress, pilotOnPress, imageSource, privacy, edited,
	showStar
} ) => (
	<HeaderView
		testID={testID}
		pilotName={pilotName}
		bodyInfo={dateToDisplay}
		optionsOnPress={optionsOnPress}
		pilotOnPress={pilotOnPress}
		imageSource={imageSource}
		bodySecondaryInfo={privacy}
		bodyTertiaryInfo={edited && 'Edited'}
		showStar={showStar}
	/>
);

PostHeaderView.propTypes = {
	testID: PropTypes.string,
	pilotName: PropTypes.string.isRequired,
	dateToDisplay: PropTypes.string.isRequired,
	optionsOnPress: PropTypes.oneOfType( [ PropTypes.func, PropTypes.oneOf( [ false ] ) ] ),
	pilotOnPress: PropTypes.func,
	imageSource: PropTypes.oneOfType( [ PropTypes.string, PropTypes.object ] ),
	privacy: PropTypes.any,
	edited: PropTypes.bool,
	showStar: PropTypes.bool
};

PostHeaderView.defaultProps = {
	testID: 'postHeaderView-component',
	optionsOnPress: null,
	pilotOnPress: () => {},
	imageSource: null,
	privacy: null,
	edited: false,
	showStar: false
};

export default PostHeaderView;
