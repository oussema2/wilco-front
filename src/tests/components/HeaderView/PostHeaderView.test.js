import * as React from 'react';
import { render } from '@testing-library/react-native';
import PostHeaderView from '../../../components/HeaderView/PostHeaderView';

describe( 'PostHeaderView', () => {
	const testID = 'testing-headerView-component';
	const pilotName = 'A pilot name';
	const dateToDisplay = '4h';

	it( 'renders the headerView', () => {
		const component = render(
			<PostHeaderView
				testID={testID}
				pilotName={pilotName}
				dateToDisplay={dateToDisplay}
			/>
		);

		expect( component.queryByTestId( testID ) ).toBeDefined();
		expect( component.queryByText( pilotName ) ).not.toBeNull();
		expect( component.queryByText( dateToDisplay ) ).not.toBeNull();
		expect( component ).toMatchSnapshot();
	} );
} );
