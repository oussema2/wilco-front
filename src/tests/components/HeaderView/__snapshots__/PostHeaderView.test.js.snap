// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostHeaderView renders the headerView 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "minHeight": 44,
    }
  }
  testID="testing-headerView-component"
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "minHeight": 44,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "aspectRatio": 1,
          "backgroundColor": "#A0A0A0",
          "borderRadius": 22,
          "height": 44,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#A0A0A0",
              "borderRadius": 68,
              "display": "flex",
              "height": 44,
              "justifyContent": "center",
            }
          }
          testID="userAvatar-image"
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "aspectRatio": 1,
                  "height": 24,
                  "width": 32,
                },
              ]
            }
          >
            <FastImageView
              onFastImageLoadEnd={[Function]}
              resizeMode="cover"
              source={
                Object {
                  "testUri": "../../../src/assets/icons/user.png",
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
              testID="image"
              tintColor="#FFFFFF"
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "aspectRatio": 1,
                  "backgroundColor": "#A0A0A0",
                  "borderRadius": 68,
                  "display": "flex",
                  "height": 44,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "aspectRatio": 1,
                      "height": 24,
                      "width": 32,
                    },
                  ]
                }
              >
                <FastImageView
                  resizeMode="cover"
                  source={
                    Object {
                      "testUri": "../../../src/assets/icons/user.png",
                    }
                  }
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                  testID="userAvatar-image-loader"
                  tintColor="#FFFFFF"
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "marginLeft": 10,
          "marginVertical": 2,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#1F2E4E",
              "flexWrap": "wrap",
              "fontFamily": "NotoSansJP-Bold",
              "fontSize": 14,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 20,
            }
          }
          testID="pilotName-text"
        >
          <Text>
            A pilot name
          </Text>
        </Text>
      </View>
      <View
        style={
          Object {
            "marginTop": 4,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#767676",
              "flexWrap": "wrap",
              "fontFamily": "NotoSansJP-Regular",
              "fontSize": 12,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 16,
            }
          }
          testID="bodyInfo-text"
        >
          4h
        </Text>
      </View>
    </View>
  </View>
</View>
`;
